File: src/android/CustomInAppMessageManagerListener.java
package com.appboy.cordova;

import android.app.Activity;
import android.view.View;
import android.widget.Toast;

import com.braze.models.inappmessage.IInAppMessage;
import com.braze.models.inappmessage.MessageButton;
import com.braze.ui.inappmessage.InAppMessageCloser;
import com.braze.ui.inappmessage.InAppMessageOperation;
import com.braze.ui.inappmessage.listeners.IInAppMessageManagerListener;

import java.util.Map;

public class CustomInAppMessageManagerListener implements IInAppMessageManagerListener {
  private final Activity mActivity;
  private Double inAppDisplayAttempts = 0;

  public CustomInAppMessageManagerListener(Activity activity) {
    mActivity = activity;
  }

  @Override
  private InAppMessageOperation beforeInAppMessageDisplayed(inAppMessage IInAppMessage) {
    if (this.inAppDisplayAttempts >= 1) {
      Log.i("Set in-app to display now");
      return InAppMessageOperation.DISPLAY_NOW;
    } else {
      Log.i("Set in-app to display now");
      return InAppMessageOperation.DISPLAY_LATER;
    }
  }
}

File: src/android/CustomInAppMessageManagerListener.kt
package com.braze.cordova

import android.app.Activity
import com.braze.ui.inappmessage.InAppMessageOperation
import com.braze.ui.inappmessage.listeners.IInAppMessageManagerListener
import com.braze.models.inappmessage.IInAppMessage
import com.braze.support.BrazeLogger.brazelog

class CustomInAppMessageManagerListener(activity: Activity) : IInAppMessageManagerListener {
  private var mActivity: Activity = activity
  var inAppDisplayAttempts = 0

  override fun beforeInAppMessageDisplayed(inAppMessage: IInAppMessage): InAppMessageOperation {
    return if (this.inAppDisplayAttempts >= 1) {
      brazelog { "Set in-app to display now"}
      InAppMessageOperation.DISPLAY_NOW
    } else {
      brazelog { "Set in-app to display later"}
      InAppMessageOperation.DISCARD
    }
  }
}